src/backend/utils/adt/agtype.c

```diff
@@ -3090,18 +3152,27 @@ Datum agtype_to_text(PG_FUNCTION_ARGS)
 
     /* Return null if arg_agt is null. This covers SQL and Agtype NULLS */
     if (arg_agt == NULL)
+    {
         PG_RETURN_NULL();
+    }
 
     /* check that we have a scalar value */
     if (!AGT_ROOT_IS_SCALAR(arg_agt))
+    {
+        PG_FREE_IF_COPY(arg_agt, 0);
```

Comment:
We don't need this here, right?

Reply:
Good catch, I thought that I removed all those. It is unnecessary.